src/NzbDrone.Core/Notifications/NotificationService.cs

```diff
@@ -9,21 +9,32 @@
 using NzbDrone.Core.Qualities;
 using NzbDrone.Core.ThingiProvider;
 using NzbDrone.Core.Tv;
+using NzbDrone.Core.HealthCheck;
+using NzbDrone.Core.HealthCheck.Checks;
+using NzbDrone.Core.Update;
 
 namespace NzbDrone.Core.Notifications
 {
     public class NotificationService
         : IHandle<EpisodeGrabbedEvent>,
           IHandle<EpisodeDownloadedEvent>,
-          IHandle<SeriesRenamedEvent>
+          IHandle<SeriesRenamedEvent>,
+          IHandleAsync<HealthCheckCompleteEvent>
     {
         private readonly INotificationFactory _notificationFactory;
         private readonly Logger _logger;
+        private readonly IHealthCheckService _healthCheckService;
+        private readonly ICheckUpdateService _checkUpdateService;
 
-        public NotificationService(INotificationFactory notificationFactory, Logger logger)
+        private UpdatePackage _lastUpdate;
```

Comment:
This is only going to be useful for the current lifecycle of the app, if Sonarr was restarted to install an update and it failed then they would get another notification the next time it ran.